app/src/main/java/com/github/se/travelpouch/ui/notifications/NotificationsScreen.kt

```diff
@@ -0,0 +1,180 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.travelpouch.model.activity.ActivityViewModel
+import com.github.se.travelpouch.model.documents.DocumentViewModel
+import com.github.se.travelpouch.model.events.EventViewModel
+import com.github.se.travelpouch.model.notifications.Notification
+import com.github.se.travelpouch.model.notifications.NotificationViewModel
+import com.github.se.travelpouch.model.profile.ProfileModelView
+import com.github.se.travelpouch.model.travels.ListTravelViewModel
+import com.github.se.travelpouch.ui.navigation.BottomNavigationMenu
+import com.github.se.travelpouch.ui.navigation.NavigationActions
+import com.github.se.travelpouch.ui.navigation.TopLevelDestinations
+import java.util.Calendar
+import java.util.Date
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun NotificationsScreen(
+    navigationActions: NavigationActions,
+    notificationViewModel: NotificationViewModel,
+    profileModelView: ProfileModelView,
+    listTravelViewModel: ListTravelViewModel,
+    activityViewModel: ActivityViewModel,
+    documentViewModel: DocumentViewModel,
+    eventsViewModel: EventViewModel
+) {
+  profileModelView.getProfile()
+
+  val profile = profileModelView.profile.collectAsState()
+  notificationViewModel.loadNotificationsForUser(profile.value.fsUid)
+  val notifications by notificationViewModel.notifications.collectAsState()
+
+  val categorizedNotifications = categorizeNotifications(notifications)
+  val context = LocalContext.current
+
+  Scaffold(
+      modifier = Modifier.testTag("ScalNotificationsScreen"),
+      topBar = {
+        TopAppBar(
+            modifier = Modifier.padding(8.dp).testTag("TopAppBarNotificationsScreen"),
+            title = {
+              Text(
+                  text = "Notifications",
+                  fontSize = 32.sp,
+                  fontWeight = FontWeight.Bold,
+                  color = Color.Black,
+                  modifier = Modifier.fillMaxWidth().testTag("TitleNotificationsScreen"))
+            },
+            actions = {
+              Button(
+                  onClick = {
+                    notificationViewModel.deleteAllNotificationsForUser(
+                        profile.value.fsUid,
+                        onSuccess = {
+                          Toast.makeText(context, "All notifications deleted", Toast.LENGTH_SHORT)
+                              .show()
+                          notificationViewModel.loadNotificationsForUser(profile.value.fsUid)
```

Comment:
I disagree with this block. If you delete all the notifications, why calling a loadNotifications ? If this is to avoid the loss of notifications that were not correctly deleted, I let you refer to my comment on the method deleteAllNotificationsForUser in the notification repository

Reply:
Once deleted notifications, it reloads all notifications to update the screen

Reply:
No, this is not useful. You delete from firebase so if you load from firebase, you get nothing each time.